provider/inmemory.go

```diff
@@ -78,13 +68,30 @@ func NewInMemoryProviderWithDomainAndLogging(domain string) *InMemoryProvider {
 			for _, v := range changes.Delete {
 				log.Infof("DELETE: %v", v)
 			}
-		},
-		OnRecords: func() {},
-		domain:    domain,
-		client:    newInMemoryClient(),
+		}
```

Comment:
yes, thought about the same, but for now I would just go with this. We can change it later, where is a real need to do it